src/txn/txn_rollback_to_stable.c

```diff
@@ -162,43 +271,50 @@ __txn_abort_newer_col_fix(
 
     /* Review the changes to the original on-page data items */
     if ((ins = WT_COL_UPDATE_SINGLE(page)) != NULL)
-        __txn_abort_newer_insert(session, ins, rollback_timestamp);
+        WT_IGNORE_RET_PTR(__txn_abort_newer_insert(session, ins, rollback_timestamp));
 
     /* Review the append list */
     if ((ins = WT_COL_APPEND(page)) != NULL)
-        __txn_abort_newer_insert(session, ins, rollback_timestamp);
+        WT_IGNORE_RET_PTR(__txn_abort_newer_insert(session, ins, rollback_timestamp));
 }
 
 /*
  * __txn_abort_newer_row_leaf --
  *     Abort updates on a row leaf page with timestamps newer than the rollback timestamp.
  */
-static void
+static int
 __txn_abort_newer_row_leaf(
   WT_SESSION_IMPL *session, WT_PAGE *page, wt_timestamp_t rollback_timestamp)
 {
     WT_INSERT_HEAD *insert;
     WT_ROW *rip;
-    WT_UPDATE *upd;
+    WT_UPDATE *insert_upd, *upd;
     uint32_t i;
 
     /*
      * Review the insert list for keys before the first entry on the disk page.
      */
     if ((insert = WT_ROW_INSERT_SMALLEST(page)) != NULL)
-        __txn_abort_newer_insert(session, insert, rollback_timestamp);
+        WT_IGNORE_RET_PTR(__txn_abort_newer_insert(session, insert, rollback_timestamp));
 
+    insert_upd = NULL;
```

Comment:
Doesn't this have to be re-initialized for every trip through the loop?
```
*** ../txn/txn_rollback_to_stable.c     Thu Jan 30 23:49:34 2020
--- -   Thu Jan 30 23:56:08 2020
*************** 
*** 297,303 ****
      if ((insert = WT_ROW_INSERT_SMALLEST(page)) != NULL)
          WT_IGNORE_RET_PTR(__txn_abort_newer_insert(session, insert, rollback_timestamp));
          
-     insert_upd = NULL;
      /*
       * Review updates that belong to keys that are on the disk image, as well as for keys inserted
       * since the page was read from disk. 
--- 297,302 ----
*************** 
*** 306,311 ****
--- 305,311 ----
          if ((upd = WT_ROW_UPDATE(page, rip)) != NULL)
              upd = __txn_abort_newer_update(session, upd, rollback_timestamp);
              
+       insert_upd = NULL;
          if ((insert = WT_ROW_INSERT(page, rip)) != NULL)
              insert_upd = __txn_abort_newer_insert(session, insert, rollback_timestamp);
```

Reply:
Yes, it needs to set NULL inside the loop. Fixed.